File: patch_server3.py
import re

with open('server.ts', 'r') as f:
    content = f.read()

content = content.replace("if (parsed._internalReasoning && !parsed._internalReasoning) { parsed._internalReasoning = parsed._internalReasoning; }", "")
content = content.replace("if (parsedData._internalReasoning && !parsedData._internalReasoning) { parsedData._internalReasoning = parsedData._internalReasoning; }", "")

with open('server.ts', 'w') as f:
    f.write(content)
print("Other assignments patched.")

File: patch_server3.ts
import fs from 'fs';
let code = fs.readFileSync('server.ts', 'utf8');

code = code.replace(
  /if \(parsedData\.nutrients && parsedData\.itemsBreakdown && parsedData\.itemsBreakdown\.length === 1 && \(userSelectedMode/g,
  'if (parsedData.nutrients && parsedData.itemsBreakdown && (userSelectedMode'
);

fs.writeFileSync('server.ts', code);
console.log("Patched itemsBreakdown length guard in server.ts");
